homeassistant/components/prometheus/__init__.py

```diff
@@ -369,6 +369,16 @@ def _handle_person(self, state):
         value = self.state_as_number(state)
         metric.labels(**self._labels(state)).set(value)
 
+    def _handle_cover(self, state):
+        metric = self._metric(
+            "cover_state",
+            self.prometheus_cli.Gauge,
+            "State of the cover (0/1)",
```

Comment:
I am not familiar with the `cover` domain, but after taking a look at [its code](https://github.com/home-assistant/core/blob/9008006ac813736be92c5ba4c84e826e6d1f8c34/homeassistant/components/cover/__init__.py#L266) I think its state is not really boolean, but rather an enum: it can have one of four values (STATE_OPENING, STATE_CLOSING, STATE_CLOSED, STATE_OPEN).

The recommended way to export such state attribute to Prometheus is to add state as a metric label, exporting `1` for the currently-active state, and `0` for all other states. For example, if the current state is "open", the library would export the following metrics:
```
cover_state{state="open"} 1
cover_state{state="opening"} 0
cover_state{state="closed"} 0
cover_state{state="closing"} 0
```

You can see examples of how this is implemented in the [hvac domain metrics](https://github.com/home-assistant/core/blob/9008006ac813736be92c5ba4c84e826e6d1f8c34/homeassistant/components/prometheus/__init__.py#L433)

Reply:
Thank you both!

I will work on these in the next few days:  the `tilt_position` and `position` properties are not always available on every cover.   I believe I have a solution in place, but I want to make sure my tests are expanded to cover the full breadth of this functionality.